tests/test-cases/Group1-Docker-Commands/1-43-Docker-CP.md

```diff
@@ -0,0 +1,80 @@
+20. Issue docker cp online:/newdir/test.txt foo.txt to the new VIC appliance
+21. Verify that test.txt is copied over to host successfully and overwrites foo.txt
+22. Issue docker cp foo.txt online:/doesnotexist/ to the new VIC appliance
+23. Issue docker cp foo.txt online:/ to the new VIC appliance
+24. Issue docker cp bar online:/ to the new VIC appliance
+25. Issue docker exec online ls / to the new VIC appliance to verify the copy operation succeeded
+26. Issue docker rm -f online to the new VIC appliance to remove online
+27. Issue docker create -i --name online -v vol1:/vol1 ${busybox} to the new VIC appliance
+28. Start online and issue docker cp bar online:/vol1/ to the new VIC appliance
+29. Issue docker exec online ls /vol1 to the new VIC appliance to verify the copy operation succeeded
+30. Issue docker rm -f online to the new VIC appliance to clean up online
+31. Issue docker create -i --name offline -v vol1:/vol1 ${busybox} to the new VIC appliance
+32. Issue docker cp foo.txt offline:/vol1 to the new VIC appliance
+33. Start offline and issue docker exec offline ls /vol1 to the new VIC appliance to verify the copy operation succeeded
+34. Issue docker rm -f offline to the new VIC appliance to clean up offline
+35. Issue docker create -i --name offline -v vol1:/vol1 -v vol2:/vol1/vol2 ${busybox} to the new VIC appliance
+36. Issue docker cp foo.txt offline:/vol1/vol2 to the new VIC appliance
+37. Start offline and issue docker exec offline ls /vol1/vol2 to the new VIC appliance to verify the copy operation succeeded
+38. Issue docker rm -f offline to the new VIC appliance to clean up offline
+39. Issue docker create -i --name offline -v vol1:/vol1 -v vol2:/vol1/vol2 -v vol3:/vol1/vol2/vol3 ${busybox} to the new VIC appliance
+40. Issue docker cp foo.txt offline:/vol1/vol2/vol3 to the new VIC appliance
+41. Start offline and issue docker exec offline ls /vol1/vol2/vol3 to the new VIC appliance to verify the copy operation succeeded
+42. Issue docker rm -f offline to the new VIC appliance to clean up offline
+43. Issue docker create -i --name online -v vol1:/vol1 ${busybox} to the new VIC appliance
+44. Issue docker create -i --name offline -v vol1:/vol1 ${busybox} to the new VIC appliance
+45. Start online and issue docker cp content offline:/vol1 to the new VIC appliance
+46. Issue docker exec online ls /vol1 to the new VIC appliance to verify the copy operation succeeded
+47. Issue docker cp offline:/vol1 . to the new VIC appliance
+48. Verify that all contents in /vol1 are copied over successfully
+49. Issue docker rm -f offline to the new VIC appliance to clean up offline
+50. Generate a file of 4 MiB and issue docker cp largefile.txt online:/vol1/ to the new VIC appliance
+51. Issue docker exec online ls -l /vol1/ to the new VIC appliance to verify the copy operation succeeded
+52. Issue docker rm -f online to the new VIC appliance to clean up online
+53. Clean up files and directories and volumes created at setup time
```

Comment:
@mhagen-vmware  I am addressing 1-3 right now, 4 can't be tested until we add in NFS data source/data sink support, and it will probably be part of the nightly test. 5 doesn't cause an error, copy to /dev/null and /dev/zeros acts like a nop. As of now there is a limitation on the serial port which restricts the amount of data that can be transferred successfully to about 7mb. This is smaller than the disk capacity we can have for a vmdk volume, so I am not sure if it's possible to test files than would exceed the limit. I can still write a test for it and gate it with an issue to fix this limitation.